### DISCUSSION TITLE
Computed property on nested object

### DISCUSSION THREAD

**Discussion by viniciusccarvalho** [quasarframework/quasar#14975] [General - Components / Directives / etc]
Hi folks, I'm wondering how can I compute a nested property inside a loop. I have a block for candidates and for each candidate I have an object with an array of skills, this is presented via sliders so users can drag and set correct value. I also have an average property I'd like to be updated to match the average of the array. However I'm not sure how to either watch this nested property (note the array of candidates may be empty some times). I though on using the events from q-slider, but it wouldn't know what candidate the event was for. Any ideas?

Codepen link: https://codepen.io/viniciusccarvalho/pen/jOKzwmW

**Comment by suleimanas** [comment]
you need the user's scores to be reactive when adding them to candidates array - i'm not sure how you would get the users data - but just as an idea:

`const { ref, created, computed } = Vue`

```
created(){
    const users = ["Alice","Mary","John"];
    
    users.forEach((name)=> {
      const skills = ref(this.rating_config.labels.map((key)=>({key, score: 0.0})));
      const average = computed(()=>(skills.value.reduce((total, next) => total + next.score, 0) / skills.value.length));
      
      this.candidates.push({name,skills,average});
    })
  },
```
